### PR TITLE
fix missing trailing slash

### LINKED ISSUE #3126: Issue with trailing slash (raw request) in latest version of nuclei
### Nuclei version:
v2.8.4

Test template -

```yaml
id: basic-raw-example

info:
  name: Test RAW Template
  author: pdteam
  severity: info

requests:
  - raw:
      - |
        GET /test/..;/..;/ HTTP/1.1
        Host: {{Hostname}}
        Origin: {{BaseURL}}
```

### Current Behavior:
nuclei make request to `https://example.com/test/..;/..;`

### Expected Behavior:
nuclei make request to `https://example.com/test/..;/..;/`

### Steps To Reproduce:
`echo https://example.com | ./nuclei -t test.yaml -debug-req`

### Anything else:
- [x] Include tests

### PR DESCRIPTION
**Pull request by tarunKoyalwar**
## Proposed changes

- Fix Missing trailing slash caused by `Path.Join`

closes #3126 

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tarunKoyalwar** [comment]
```sh
$ go run . -u https://scanme.sh -t ~/test/raw.yaml -debug-req

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.4

		projectdiscovery.io

[INF] Using Nuclei Engine 2.8.4 (latest)
[INF] Using Nuclei Templates 9.3.3 (latest)
[INF] Templates added in last update: 238
[INF] Templates loaded for scan: 1
[INF] Targets loaded for scan: 1
[INF] [basic-raw-example] Dumped HTTP request for https://scanme.sh/test/..;/..;/

GET /test/..;/..;/ HTTP/1.1
Host: scanme.sh
User-Agent: Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/37.0.2049.0 Safari/537.36
Connection: close
Origin: https://scanme.sh
Accept-Encoding: gzip

[INF] No results found. Better luck next time!

```

**Comment by tarunKoyalwar** [comment]
# Cause

- This was a edgecase where empty path was passed to `Path.Join` 
- Path.Join internally calls `Path.Clean` which filters out relative paths urls or other normalization 

> https://github.com/golang/go/blob/db36eca33c389871b132ffb1a84fd534a349e8d8/src/path/path.go#L51-L136

**Review by ehsandeep — APPROVED** [review]

File: integration_tests/http/raw-path.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+id: basic-raw-example
+
+info:
+  name: Test RAW Template
```
Comment (Mzack9999) [inline integration_tests/http/raw-path.yaml:4]: The `id` and `name` should mention trailing forward slash. I'd suggest something like this:
```
id: raw-with-path-final-forward-slash
name: Test Raw Request with a final forward slash in the path
```

File: v2/cmd/integration-test/http.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ var httpTestcases = map[string]testutils.TestCase{
 	"http/raw-dynamic-extractor.yaml":               &httpRawDynamicExtractor{},
 	"http/raw-get-query.yaml":                       &httpRawGetQuery{},
 	"http/raw-get.yaml":                             &httpRawGet{},
+	"http/raw-path.yaml":                            &httpRawPath{},
```
Comment (Mzack9999) [inline v2/cmd/integration-test/http.go:36]: same as before, the filename should mention the forward slash, along with the test name

File: v2/cmd/integration-test/http.go [hunk 3]
```diff
@@ -505,6 +506,31 @@ func (h *httpRawGet) Execute(filePath string) error {
 	return expectResultsCount(results, 1)
 }
 
+type httpRawPath struct{}
```
Comment (Mzack9999) [inline v2/cmd/integration-test/http.go:509]: see my previous comment

File: v2/cmd/integration-test/http.go [hunk 4]
```diff
@@ -505,6 +506,31 @@ func (h *httpRawGet) Execute(filePath string) error {
...
+
+func (h *httpRawPath) Execute(filepath string) error {
+	router := httprouter.New()
+	var routerErr error
+
+	router.GET("/", func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+		if r.RequestURI != "/test/..;/..;/" {
```
Comment (Mzack9999) [inline v2/cmd/integration-test/http.go:516]: Similarly to https://github.com/projectdiscovery/nuclei/blob/16d15e216c70044917d8d3180e054a019ed430d4/v2/cmd/integration-test/http.go#L468 I think we can return some expected message and match it with template matchers.

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
suggesting some minor change

**Review by Mzack9999 — APPROVED** [review]

**Pull request merged**